Source/WebCore/crypto/mac/CryptoKeyECMac.cpp

```diff
@@ -254,49 +313,51 @@ RefPtr<CryptoKeyEC> CryptoKeyEC::platformImportSpki(CryptoAlgorithmIdentifier id
-        if (CCECCryptorImportKey(kCCImportKeyBinary, keyData.data() + index, keyData.size() - index, ccECKeyPublic, &ccPublicKey))
+    if (!doesUncompressedPointMatchNamedCurve(curve, keyData.size() - index)) {
+        if (CCECCryptorImportKey(kCCImportKeyCompact, keyData.data() + index+1, keyData.size() - index-1, ccECKeyPublic, &ccPublicKey))
             return nullptr;
     } else {
-        ++index;
-        if (CCECCryptorImportKey(kCCImportKeyCompact, keyData.data() + index, keyData.size() - index, ccECKeyPublic, &ccPublicKey))
+        if (CCECCryptorImportKey(kCCImportKeyBinary, keyData.data() + index, keyData.size() - index, ccECKeyPublic, &ccPublicKey))
             return nullptr;
-        
     }
-    return create(identifier, curve, CryptoKeyType::Public, PlatformECKeyContainer(ccPublicKey), extractable, usages);
+    return create(identifier, curve, CryptoKeyType::Public, std::unique_ptr<typename std::remove_pointer<CCECCryptorRef>::type, WebCore::CCECCryptorRefDeleter>(ccPublicKey), extractable, usages);
 }
 
 Vector<uint8_t> CryptoKeyEC::platformExportSpki() const
 {
     size_t expectedKeySize = 2 * keySizeInBytes() + 1; // Per Section 2.3.4 of http://www.secg.org/sec1-v2.pdf
     Vector<uint8_t> keyBytes(expectedKeySize);
     size_t keySize = keyBytes.size();
-    if (UNLIKELY(CCECCryptorExportKey(kCCImportKeyBinary, keyBytes.data(), &keySize, ccECKeyPublic, m_platformKey.get()) || keySize != expectedKeySize))
-        return { };
-
-    // The following addes SPKI header to a raw EC public key.
-    // Once the underlying crypto library is updated to output SPKI EC Key, we should remove this hack.
-    // <rdar://problem/30987628>
-    const uint8_t* oid;
-    size_t oidSize = getOID(namedCurve(), oid);
-
-    // SEQUENCE + length(1) + OID id-ecPublicKey + OID secp256r1/OID secp384r1/OID secp521r1 + BIT STRING + length(?) + InitialOctet + Key size
-    size_t totalSize = sizeof(IdEcPublicKey) + oidSize + bytesNeededForEncodedLength(keySize + 1) + keySize + 4;
-
     Vector<uint8_t> result;
-    result.reserveInitialCapacity(totalSize + bytesNeededForEncodedLength(totalSize) + 1);
-    result.append(SequenceMark);
-    addEncodedASN1Length(result, totalSize);
-    result.append(SequenceMark);
-    addEncodedASN1Length(result, sizeof(IdEcPublicKey) + oidSize);
-    result.append(IdEcPublicKey, sizeof(IdEcPublicKey));
-    result.append(oid, oidSize);
-    result.append(BitStringMark);
-    addEncodedASN1Length(result, keySize + 1);
-    result.append(InitialOctet);
-    result.append(keyBytes.data(), keyBytes.size());
-
-    return result;
+    auto keyBytesdata = keyBytes.data();
+    return WTF::switchOn( m_platformKey, [&](const std::unique_ptr<typename std::remove_pointer<CCECCryptorRef>::type, WebCore::CCECCryptorRefDeleter>& options) mutable->Vector<uint8_t> {
+        if (UNLIKELY(CCECCryptorExportKey(kCCImportKeyBinary, keyBytesdata , &keySize, ccECKeyPublic, options.get()) || keySize != expectedKeySize))
```

Comment:
keyBytesdata -> keyBytesData?

Also, you have an extra space after keyBytesdata.

Reply:
changed to auto&